test/linalg/symmetric.jl

```diff
@@ -42,206 +42,253 @@ end
+                end
+            end
+            @testset "similar" begin
+                @test isa(similar(Symmetric(asym)), Symmetric{eltya})
+                @test isa(similar(Hermitian(aherm)), Hermitian{eltya})
+                @test isa(similar(Symmetric(asym), Int), Symmetric{Int})
+                @test isa(similar(Hermitian(aherm), Int), Hermitian{Int})
+                @test isa(similar(Symmetric(asym), (3,2)), Matrix{eltya})
+                @test isa(similar(Hermitian(aherm), (3,2)), Matrix{eltya})
+                @test isa(similar(Symmetric(asym), Int, (3,2)), Matrix{Int})
+                @test isa(similar(Hermitian(aherm), Int, (3,2)), Matrix{Int})
+            end
+            @testset "full" begin
+                @test asym  == full(Symmetric(asym))
+                @test aherm == full(Hermitian(aherm))
+            end
 
-        # constructor
-        @test Symmetric(Symmetric(asym, :U))     === Symmetric(asym, :U)
-        @test Hermitian(Hermitian(asym, :U))     === Hermitian(asym, :U)
-        @test Symmetric(Symmetric(asym, :U), :U) === Symmetric(asym, :U)
-        @test Hermitian(Hermitian(asym, :U), :U) === Hermitian(asym, :U)
-        @test_throws ArgumentError Symmetric(Symmetric(asym, :U), :L)
-        @test_throws ArgumentError Hermitian(Hermitian(asym, :U), :L)
-        # mixed cases with Hermitian/Symmetric
-        @test Symmetric(Hermitian(asym, :U))     === Symmetric(asym, :U)
-        @test Hermitian(Symmetric(asym, :U))     === Hermitian(asym, :U)
-        @test Symmetric(Hermitian(asym, :U), :U) === Symmetric(asym, :U)
-        @test Hermitian(Symmetric(asym, :U), :U) === Hermitian(asym, :U)
-        @test_throws ArgumentError Symmetric(Hermitian(asym, :U), :L)
-        @test_throws ArgumentError Hermitian(Symmetric(asym, :U), :L)
-
-        # similar
-        @test isa(similar(Symmetric(asym)), Symmetric{eltya})
-        @test isa(similar(Hermitian(asym)), Hermitian{eltya})
-        @test isa(similar(Symmetric(asym), Int), Symmetric{Int})
-        @test isa(similar(Hermitian(asym), Int), Hermitian{Int})
-        @test isa(similar(Symmetric(asym), (3,2)), Matrix{eltya})
-        @test isa(similar(Hermitian(asym), (3,2)), Matrix{eltya})
-        @test isa(similar(Symmetric(asym), Int, (3,2)), Matrix{Int})
-        @test isa(similar(Hermitian(asym), Int, (3,2)), Matrix{Int})
-
-        # full
-        @test asym == full(Hermitian(asym))
-
-        # parent
-        @test asym == parent(Hermitian(asym))
-
-        # getindex
-        @test asym[1,1] == Hermitian(asym)[1,1]
-        @test asym[1,1] == Symmetric(asym)[1,1]
-
-        #trace
-        @test trace(asym) == trace(Hermitian(asym))
-
-        # issymmetric, ishermitian
-        if eltya <: Real
-            @test issymmetric(Symmetric(asym))
-            @test ishermitian(Symmetric(asym))
-        end
-        if eltya <: Complex
-            @test ishermitian(Symmetric(b + b'))
```

Comment:
this one might be kind of useful to keep in addition to `aherm`, since `b` is complex but with zero imaginary component